pkg/sql/opt_catalog.go

```diff
@@ -1062,7 +1062,7 @@ func (ot *optTable) WritableIndexCount() int {
 // DeletableIndexCount is part of the cat.Table interface.
 func (ot *optTable) DeletableIndexCount() int {
 	// Primary index is always present, so count is always >= 1.
-	return len(ot.desc.AllIndexes())
+	return len(ot.desc.DeletableNonPrimaryIndexes()) + 1
```

Comment:
Why this change? This is the same thing.

Reply:
I get what you're saying and your change here makes sense in that light. That being said I just pulled `master` I'm not seeing what you describe in the current implementation of `AllIndexes()`. Extending the `indexCache` struct to support backfilling indexes seems like it would lie within the scope of this PR.

Reply:
My comment above is incorrect, sorry. You did what was required as far as I'm concerned.